pkg/e2e/compose_environment_test.go

```diff
@@ -27,12 +27,39 @@ import (
 func TestEnvPriority(t *testing.T) {
 	c := NewParallelCLI(t)
 
-	projectDir := "./fixtures/environment/env-priority"
-
 	t.Run("up", func(t *testing.T) {
 		c.RunDockerOrExitError(t, "rmi", "env-compose-priority")
 		c.RunDockerComposeCmd(t, "-f", "./fixtures/environment/env-priority/compose-with-env.yaml",
-			"--project-directory", projectDir, "up", "-d", "--build")
+			"up", "-d", "--build")
+	})
+
+	// Full options activated
+	// 1. Compose file
+	// 2. Shell environment variables <-- Result expected
+	// 3. Environment file
+	// 4. Dockerfile
+	// 5. Variable is not defined
+	t.Run("compose file priority", func(t *testing.T) {
+		cmd := c.NewDockerComposeCmd(t, "-f", "./fixtures/environment/env-priority/compose-with-env.yaml",
+			"--env-file", "./fixtures/environment/env-priority/.env.override",
+			"run", "--rm", "-e", "WHEREAMI", "env-compose-priority")
```

Comment:
So there's a difference between `-e WHEREAMI` with the var defined, and `-e WHEREAMI=shell`? In this case shouldn't `3.` be the expected result?

Reply:
In the first, the variable value will be inherited from the environment. 
The second one will have the explicit value passed (`shell` in this case).